pkg/observability/tracing.go

```diff
@@ -0,0 +1,62 @@
+package observability
+
+import (
+	"context"
+	"log"
+
+	"github.com/gobuffalo/buffalo"
+	"go.opencensus.io/exporter/jaeger"
+	"go.opencensus.io/trace"
+)
+
+// ObserverContext is a context that wraps around a buffaloCtx and also keeps a span context
+// for spans to use it
+type ObserverContext struct {
+	buffalo.Context
+	SpanCtx context.Context
+}
+
+// RegisterTraceExporter returns a jaeger exporter for exporting traces to opencensus.
+// It should in the future have a nice sampling rate defined
+func RegisterTraceExporter(service string) *(jaeger.Exporter) {
+	collectorEndpointURI := "http://0.0.0.0:14268"
+
+	je, err := jaeger.NewExporter(jaeger.Options{
+		Endpoint:    collectorEndpointURI,
+		ServiceName: service,
+	})
+	if err != nil {
+		log.Fatalf("Failed to create the Jaeger exporter: %v", err)
+	}
+
+	// And now finally register it as a Trace Exporter
+	trace.RegisterExporter(je)
+	trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
+	return je
+}
+
+// Tracer is a middleware that starts a span from the top of a buffalo context
+// and propates it to the bottom of the stack
+func Tracer(service string) buffalo.MiddlewareFunc {
```

Comment:
Yes, we need traces inside all the handlers. The reason being you want to see what are the child handlers. Opencensus is pretty cool because you do not have to specify a ChildSpan function like opentracing, but just say StartSpan. It will create a child span if it already exists, otherwise it will create a parent span.

I think we are on the same page as far as the tracing goes for the packages and modules.